src/UI/Dock.re

```diff
@@ -5,17 +5,16 @@ open Oni_Model;
 
 let button = Style.[marginVertical(24)];
 
-let toggleExplorer = ({fileExplorer, _}: State.t, _) => {
-  let action =
-    fileExplorer.isOpen
-      ? Actions.RemoveDockItem(WindowManager.ExplorerDock)
-      : Actions.AddDockItem(WindowManager.ExplorerDock);
-  GlobalContext.current().dispatch(action);
+let onExplorerClick = _ => {
+  GlobalContext.current().dispatch(Actions.ActivityBar(FileExplorerClick));
 };
 
-let toggleSearch = ({searchPane, _}: State.t, _) => {
-  let action = searchPane == None ? Actions.SearchShow : Actions.SearchHide;
-  GlobalContext.current().dispatch(action);
+let onSearchClick = _ => {
+  GlobalContext.current().dispatch(Actions.ActivityBar(SearchClick));
+};
+
+let onExtensionsClick = _ => {
+  GlobalContext.current().dispatch(Actions.ActivityBar(ExtensionsClick));
```

Comment:
This was related to some of the feedback on #1034 - wanted to run it by you @glennsl